coderd/templateversions.go

```diff
@@ -266,6 +270,118 @@ func (api *API) patchCancelTemplateVersion(rw http.ResponseWriter, r *http.Reque
 	})
 }
 
+// @Summary Open dynamic parameters WebSocket by template version
+// @ID open-dynamic-parameters-websocket-by-template-version
+// @Security CoderSessionToken
+// @Tags Templates
+// @Param templateversion path string true "Template version ID" format(uuid)
+// @Success 101
+// @Router /templateversions/{templateversion}/dynamic-parameters [get]
+func (api *API) templateVersionDynamicParameters(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	templateVersion := httpmw.TemplateVersionParam(r)
+
+	// Check that the job has completed successfully
+	job, err := api.Database.GetProvisionerJobByID(ctx, templateVersion.JobID)
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error fetching provisioner job.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+	if !job.CompletedAt.Valid {
+		httpapi.Write(ctx, rw, http.StatusTooEarly, codersdk.Response{
+			Message: "Job hasn't completed!",
+		})
+		return
+	}
+
+	// Having the Terraform plan available for the evaluation engine is helpful
+	// for populating values from data blocks, but isn't strictly required. If
+	// we don't have a cached plan available, we just use an empty one instead.
+	var plan json.RawMessage = []byte("{}")
+	tf, err := api.Database.GetTemplateVersionTerraformValues(ctx, templateVersion.ID)
+	if err == nil {
+		plan = tf.CachedPlan
+	}
+
+	input := preview.Input{
+		PlanJSON:        plan,
+		ParameterValues: map[string]string{},
+		// TODO: fill this out
```

Comment:
I'm gonna take care of this in a separate pr

Reply:
When you do, it might be nice to have all of this in a seperate pkg in `coder/coder`.  All the code for taking a database & pulling all the right records. Then we can write unit tests without having to invoke the api.

Reply:
Can we just elaborate the `TODO` for now?
```
// TODO: populate the `Owner` field with the user's coder context. These values will be
// used for the values of `coder_*` terraform blocks.
```